glean_parser/util.py

```diff
@@ -397,20 +414,32 @@ def is_expired(expires: str) -> bool:
         return False
     elif expires == "expired":
         return True
+    elif major_version is not None:
+        return parse_expiration_version(expires) >= major_version
     else:
-        date = parse_expires(expires)
+        date = parse_expiration_date(expires)
         return date <= datetime.datetime.utcnow().date()
 
 
-def validate_expires(expires: str) -> None:
+def validate_expires(expires: str, major_version: Optional[int] = None) -> None:
     """
-    Raises a ValueError in case the `expires` is not ISO8601 parseable,
-    or in case the date is more than 730 days (~2 years) in the future.
+    If expiration by major version is enabled, raises a ValueError in
+    case `expires` is not a positive integer.
+    Otherwise raises a ValueError in case the `expires` is not ISO8601
+    parseable, or in case the date is more than 730 days (~2 years) in
+    the future.
     """
     if expires in ("never", "expired"):
         return
 
-    date = parse_expires(expires)
+    if major_version is not None:
+        parse_expiration_version(expires)
+        # Don't need to keep parsing dates if expiration by version
+        # is enabled. We don't allow mixing dates and versions for a
+        # single product.
```

Comment:
I agree with this restriction.  Should we also document this in the description in the schema above (which ends up in the docs)?

Reply:
Good call.